src/Shared/EnvironmentUtilities.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Build.Shared
+{
+    internal static partial class EnvironmentUtilities
+    {
+        public static bool Is64BitProcess => Marshal.SizeOf<IntPtr>() == 8;
+
+        //  Copied from .NET Framework source for Environment.Is64BitOperatingSystem (along with Win32Native APIs)
+        public static bool Is64BitOperatingSystem
+        {
+            get
+            {
+                bool isWow64; // WinXP SP2+ and Win2k3 SP1+
+                return Win32Native.DoesWin32MethodExist(Win32Native.KERNEL32, "IsWow64Process")
```

Comment:
It does require an additional unnecessary p/invoke everytime this method is accessed though. The original code was also hardcoded to `true` in the `!WIN32` case, we could avoid all the p/invokes by checking `Is64BitProcess == true` here since that can only be true on 64bit OSes.

But yeah maybe that's micro-optimization and we should just wait until corefx adds `Is64BitOperatingSystem` back: https://github.com/dotnet/corefx/issues/999 :smile: